### PR TITLE
docs: add ability to filter by category in sidebar

### PR DESCRIPTION
**Pull request by lesha1201**
This PR adds ability to filter in the sidebar (navbar) by a category like on Lodash docs or Ramda docs.

https://github.com/remeda/remeda/assets/10157660/7c7b8f00-2ba1-4910-b258-52c9ad296632

### CODE REVIEW

File: docs/src/components/navbar.tsx [hunk 1]
```diff
@@ -31,9 +31,11 @@ export function Navbar({
         ([category, entries]) =>
           [
             category,
-            entries.filter(({ name }) =>
-              name.toLowerCase().includes(lowerCaseQuery),
-            ),
+            category.toLowerCase().includes(lowerCaseQuery)
```
Comment (lesha1201) [inline docs/src/components/navbar.tsx:34]: Not sure but `startsWith` might be better for this case.

Comment (eranhirsch) [reply docs/src/components/navbar.tsx:34]: I agree with this observation, I think startsWith will make more sense here.

**Review by eranhirsch — CHANGES_REQUESTED** [review]
Let's use startsWith

File: docs/src/components/navbar.tsx (hunk 1 above)
Comment (lesha1201) [reply docs/src/components/navbar.tsx]: Done

**Review by eranhirsch — APPROVED** [review]
Thanks for helping us!

**Pull request merged**